patches/minecraft/net/minecraft/client/renderer/FluidBlockRenderer.java.patch

```diff
@@ -51,7 +51,17 @@
              flag7 = true;
           }
  
-@@ -239,15 +240,15 @@
+@@ -223,8 +224,7 @@
+                BlockPos blockpos = p_228796_2_.func_177972_a(direction);
+                TextureAtlasSprite textureatlassprite2 = atextureatlassprite[1];
+                if (!flag) {
+-                  Block block = p_228796_1_.func_180495_p(blockpos).func_177230_c();
+-                  if (block == Blocks.field_150359_w || block instanceof StainedGlassBlock) {
++                  if (p_228796_1_.func_180495_p(blockpos).shouldUseWaterOverlay(p_228796_1_, blockpos)) {
```

Comment:
I wonder if there would be a benefit on calling this shouldDisplayFluidOverlay, and include the IFluidState seen in the world, as a parameter?

Reply:
I can name it that if it is preferred, I wasn't really sure on what I should name it.